storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -346,53 +400,104 @@
+               (> min-replication-count @current-replication-count-conf)
+               (or (= -1 max-replication-wait-time)
+                 (< @total-wait-time max-replication-wait-time)))
         (sleep-secs 1)
         (log-debug "waiting for desired replication to be achieved.
           min-replication-count = " min-replication-count  " max-replication-wait-time = " max-replication-wait-time
-          "current-replication-count = " @current-replication-count " total-wait-time " @total-wait-time)
+          (if (not (local-mode? conf))"current-replication-count for jar key = " @current-replication-count-jar)
+          "current-replication-count for code key = " @current-replication-count-code
+          "current-replication-count for conf key = " @current-replication-count-conf
+          " total-wait-time " @total-wait-time)
         (swap! total-wait-time inc)
-        (reset! current-replication-count  (.getReplicationCount (:code-distributor nimbus) storm-id))))
-  (if (< min-replication-count @current-replication-count)
-    (log-message "desired replication count "  min-replication-count " achieved,
-      current-replication-count" @current-replication-count)
-    (log-message "desired replication count of "  min-replication-count " not achieved but we have hit the max wait time "
-      max-replication-wait-time " so moving on with replication count = " @current-replication-count)
-    )))
-
-(defn- read-storm-topology [conf storm-id]
-  (let [stormroot (master-stormdist-root conf storm-id)]
-    (Utils/deserialize
-      (FileUtils/readFileToByteArray
-        (File. (master-stormcode-path stormroot))
-        ) StormTopology)))
+        (if (not (local-mode? conf))(reset! current-replication-count-conf  (get-blob-replication-count (master-stormconf-key storm-id))))
+          (reset! current-replication-count-code  (get-blob-replication-count (master-stormcode-key storm-id)))
+          (reset! current-replication-count-jar  (get-blob-replication-count (master-stormjar-key storm-id)))))
+    (if (and (< min-replication-count @current-replication-count-conf)
+          (< min-replication-count @current-replication-count-code)
+          (< min-replication-count @current-replication-count-jar))
+      (log-message "desired replication count of "  min-replication-count " not achieved but we have hit the max wait time "
+        max-replication-wait-time " so moving on with replication count for conf key = " @current-replication-count-conf
+        " for code key = " @current-replication-count-code "for jar key = " @current-replication-count-jar)
+      (log-message "desired replication count "  min-replication-count " achieved,
+        current-replication-count for conf key " @current-replication-count-conf ",
+        current-replication-count for code key = " @current-replication-count-code ",
+        current-replication-count for jar key = " @current-replication-count-jar))))
```

Comment:
What happens if we do not close the string literal after `" achieved,` and continue it on the next line with indentation? Does all of that white-space end up getting output?

Also a nitpick: no `=` sign in `conf key "` while other two have one.